pxc-57/entrypoint.sh

```diff
@@ -98,62 +108,61 @@ fi
 		echo
 		echo 'MySQL init process done. Ready for start up.'
 		echo
-		#mv /etc/my.cnf $DATADIR
 	fi
-	touch $DATADIR/init.ok
 	chown -R mysql:mysql "$DATADIR"
 
-if [ -z "$DISCOVERY_SERVICE" ]; then
-	cluster_join=$CLUSTER_JOIN
-else
-
-echo
-echo 'Registering in the discovery service'
-echo
-
-function join {
-  local IFS="$1"
-  shift
-  joined=$(tr "$IFS" '\n' <<< "$*" | sort -un | tr '\n' "$IFS")
-  echo "${joined%?}"
-}
-
-# Read the list of registered IP addresses
-set +e
-
-ipaddr=$(hostname -i | awk ' { print $1 } ')
-hostname=$(hostname)
-
-curl http://$DISCOVERY_SERVICE/v2/keys/pxc-cluster/queue/$CLUSTER_NAME -XPOST -d value=$ipaddr -d ttl=60
-
-#get list of IP from queue 
-i=$(curl http://$DISCOVERY_SERVICE/v2/keys/pxc-cluster/queue/$CLUSTER_NAME | jq -r '.node.nodes[].value')
-
-# this remove my ip from the list
-i1="${i[@]/$ipaddr}"
-cluster_join1=$(join , $i1)
-
-# Register the current IP in the discovery service
-
-# key set to expire in 30 sec. There is a cronjob that should update them regularly
-curl http://$DISCOVERY_SERVICE/v2/keys/pxc-cluster/$CLUSTER_NAME/$ipaddr/ipaddr -XPUT -d value="$ipaddr" -d ttl=30
-curl http://$DISCOVERY_SERVICE/v2/keys/pxc-cluster/$CLUSTER_NAME/$ipaddr/hostname -XPUT -d value="$hostname" -d ttl=30
-curl http://$DISCOVERY_SERVICE/v2/keys/pxc-cluster/$CLUSTER_NAME/$ipaddr -XPUT -d ttl=30 -d dir=true -d prevExist=true
-
-#i=`curl http://$DISCOVERY_SERVICE/v2/keys/pxc-cluster/$CLUSTER_NAME/ | jq -r '.node.nodes[].value'`
-i=$(curl http://$DISCOVERY_SERVICE/v2/keys/pxc-cluster/$CLUSTER_NAME/?quorum=true | jq -r '.node.nodes[]?.key' | awk -F'/' '{print $(NF)}')
-# this remove my ip from the list
-i2="${i[@]/$ipaddr}"
-cluster_join2=$(join , $i1)
-cluster_join=$(join , $i1 $i2 )
-echo "Joining cluster $cluster_join"
-
+#--log-error=${DATADIR}error.log
+# run in standalone mode or as a cluster
+if [ !  -z "$CLUSTER_NAME" ]; then
+    containerId=$(cat /proc/self/cgroup |  awk -F  ":" '/pids/ {print $3}' | awk -F  "/" '{print $3}');
+    serviceName=$(curl -s --unix-socket /var/run/docker.sock \
```

Comment:
just found this tutorial and I will check if we can use that instead so that we don't have to use the  docker socket at all.

https://docs.docker.com/engine/swarm/networking/

I hope we can use that , looks a lot cleaner